app/src/examples/RuntimeTests/ReanimatedRuntimeTestsRunner/TestRunner.ts

```diff
@@ -280,7 +310,39 @@ export class TestRunner {
       await testSuite.beforeEach();
     }
 
-    await testCase.run();
+    if (testCase.decorator === TestDecorator.FAILING || testCase.decorator === TestDecorator.WARN) {
+      const consoleTrackerRef = testCase.decorator === TestDecorator.FAILING ? 'console.error' : 'console.warn';
+      const message = makeMutable('');
+
+      const newConsoleFuncJS = (warning: string) => {
+        this.callTracker(consoleTrackerRef);
+        message.value = warning.split('\n\nThis error is located at:')[0];
+      };
+      console.error = newConsoleFuncJS;
+      console.warn = newConsoleFuncJS;
```

Comment:
Do you ever restore original implementation of `console.warn`?

Reply:
No, since you have to reload the app to re-run the tests and it resets it.